### PR TITLE
More AWS Lambda improvements and spec changes

### PR DESCRIPTION
**Pull request by basepi**
Better error handling for the `Xxx` conversion, and remove case-insensitive `statusCode` handling, as anything besides `statusCode` is invalid and will throw a server error.

Edit: Also added the field changes from spec PR here: https://github.com/elastic/apm/pull/540

### CODE REVIEW

File: elasticapm/contrib/serverless/aws.py [hunk 1]
```diff
@@ -171,9 +167,18 @@ def __exit__(self, exc_type, exc_val, exc_tb):
...
+                "response",
+            )
             if status_code:
-                result = "HTTP {}xx".format(int(status_code) // 100)
-                elasticapm.set_transaction_result(result, override=False)
+                try:
+                    result = "HTTP {}xx".format(int(status_code) // 100)
+                    elasticapm.set_transaction_result(result, override=False)
+                except ValueError:
+                    elasticapm.set_transaction_result("HTTP 2xx", override=False)
```
Comment (beniwohli) [inline elasticapm/contrib/serverless/aws.py:181]: could you add a comment here lining out what causes the `ValueError`?

Comment (basepi) [reply elasticapm/contrib/serverless/aws.py:181]: Done. Casting to an `int` throws a `ValueError` if it's unsuccessful.

**Comment by basepi** [comment]
Thanks to @trentm for pointing out my bad assumption about case-sensitivity of `statusCode`. This is now ready for review again.

File: elasticapm/contrib/serverless/aws.py [hunk 2]
```diff
@@ -65,7 +65,7 @@ def handler(event, context):
             return {"statusCode": r.status_code, "body": "Success!"}
     """
 
-    def __init__(self, name=None, **kwargs):
+    def __init__(self, name: str = None, **kwargs) -> None:
```
Comment (beniwohli) [inline elasticapm/contrib/serverless/aws.py:68]: this is type `Optional[str]` if its default value is `None`

**Review by beniwohli — APPROVED** [review]
just a tiny typing nit, otherwise LGTM

File: elasticapm/contrib/serverless/aws.py (hunk 2 above)
Comment (basepi) [reply elasticapm/contrib/serverless/aws.py:68]: 🤦‍♂️ Honestly don't know how I missed that.

**Pull request merged**